pubsub/google/cloud/pubsub_v1/publisher/client.py

```diff
@@ -130,16 +135,40 @@ def batch(self, topic, create=False, autocommit=True):
                     create = True
 
             if create:
+                if batch_settings is None:
+                    batch_settings = self.batch_settings
+
                 batch = self._batch_class(
                     autocommit=autocommit,
                     client=self,
-                    settings=self.batch_settings,
+                    settings=batch_settings,
                     topic=topic,
                 )
                 self._batches[topic] = batch
 
         return batch
 
+    def batch(self, topic, create=False, autocommit=True):
+        """Return the current batch for the provided topic.
+
+        This will create a new batch if ``create=True`` or if no batch
+        currently exists.
+
+        Args:
+            topic (str): A string representing the topic.
+            create (bool): Whether to create a new batch. Defaults to
+                :data:`False`. If :data:`True`, this will create a new batch
+                even if one already exists.
+            autocommit (bool): Whether to autocommit this batch. This is
+                primarily useful for debugging and testing, since it allows
+                the caller to avoid some side effects that batch creation
+                might have (e.g. spawning a worker to publish a batch).
+
+        Returns:
+            ~.pubsub_v1.batch.Batch: The batch object.
+        """
+        return self._batch(topic, create, autocommit)
+
```

Comment:
What is the point of the wrapper method here?  ISTM we could have just made `batch` take an optional `batch_settings` argument.

Reply:
Luke mentioned about not allowing users to mess with the batch_settings.